config/mixpanel.py

```diff
@@ -0,0 +1,35 @@
+import os
+import flask
+
+from app.models.user import User
+from mixpanel import Mixpanel  # type: ignore
+
+
+class NotifyMixpanel:
+    def __new__(klazz, user: User = None):
+        if not os.environ.get("MIXPANEL_PROJECT_TOKEN"):
+            flask.current_app.logger.warning(
+                "MIXPANEL_PROJECT_TOKEN is not set. Mixpanel features will not be supported."
+                "In order to enable Mixpanel, set MIXPANEL_PROJECT_TOKEN environment variable."
+            )
+
+        return super(NotifyMixpanel, klazz).__new__(klazz)
+
+    def __init__(self, user: User = None) -> None:
+        super().__init__()
+
+        self.mixpanel: Mixpanel = Mixpanel(os.environ.get("MIXPANEL_PROJECT_TOKEN", ""))
+        self.user = user
+
+    def track_event(self, msg="Notify: Sent message") -> None:
```

Comment:
@jimleroyer if I had written this as 
```suggestion
    def track_login_event(self, msg="Notify: Logged in ") -> None:
```

Then the user guard I suppose would be a bit more intentional??

Reply:
The issue is not the messaging, but rather in which inconsistent state can a class end up to be. If the user is required, making it required at the function level by defining it as a function param instead of a class param would be more consistent. There would be less chance for the class to go in a wrong state in the future. But because we say the user state is global to the class but it might be optional (and possible not required for all tracking events in the future), then the internal state has more chance to be incorrect at some point. Removing the user state at the class level would remove this likelihood and would not require manual user guard in every functions using it.